### ISSUE TITLE
Make tags searchable with "#" prefix

### ISSUE THREAD

**Issue by liaprins** [liaprins/curiositycoloredglasses.com#55]
Currently search does not look at tag content. 

Also, would need to get search to ignore "#" if typed into search bar...

**Comment by liaprins** [comment]
Update: search does look at tags. So I just need to get it to ignore #s.

**Comment by liaprins** [comment]
https://forum.getkirby.com/t/fuzzy-search-plugin/9102
https://github.com/pedroborges/kirby-fuzzy-search

**Comment by liaprins** [comment]
```
<!-- TESTING FUZZY SEARCH -->
<!-- <form action="<?= search($pages->index(), '$query', 'title|text') ?>" id="navsearchform"> -->
<!-- <form action="<?= fuzzySearch($pages->index(), '$query', 'title|text') ?>" id="navsearchform"> -->
```

**Comment by liaprins** [comment]
https://forum.getkirby.com/t/fuzzy-search-plugin/9102/8
https://documentation.mamp.info/en/MAMP-Mac/FAQ/
http://localhost:8888/MAMP/index.php?language=English&page=phpinfo
https://stackoverflow.com/questions/16783558/how-can-i-add-additional-php-versions-to-mamp

**Comment by liaprins** [comment]
Progress: by changing line 44 of search.php in plugins folder to:
`if($matches = preg_match_all('!\b' . ltrim(preg_quote($query), '#') . '\b!i', $data[$key], $r)) {`
from:
`if($matches = preg_match_all('!\b' . preg_quote($query) . '\b!i', $data[$key], $r)) {`
I was able to search for #lollipop and return a post with the tag of #lollipop, and also maintain the search bar entry with the # in it.

...However, if I search for #mountains it will also return posts who just have the word mountain mentioned but not as a tag with # in front. This should only happen if I search for "mountain".

Need to try to get the above code to just apply to the tags field...

**Comment by liaprins** [comment]
More discovery, but not as good as original progress above:

changing ~line 15 in site/controllers/search.php to:
`$results = search($pages->index()->visible()->not('template', '!=', 'blog', 'comment', 'comments', 'hello', 'archive', 'library'), ltrim($query, '#'), array('words' => true, 'fields' => ['tags'] ));`
from: 
`$results = search($pages->index()->visible()->not('template', '!=', 'comment', 'comments', 'about', 'archive', 'library'), $query, array('words' => true, 'fields' => ['title', 'text', 'intro', 'tags', 'text', 'caption'] ));`
returns ONLY results with tags who match a search with a "#" preceding, and no other results.

- [ ] Plan is use the experiment in the comment above, for site/plugins/search.php, changing to:
`if($matches = preg_match_all('!\b' . ltrim(preg_quote($query), '#') . '\b!i', $data[$key], $r)) {` This way the regular search results will also work. And it's not such a problem if a search for #mountains returns results that mention the word "mountains" in their text, but not explicitly as a tag; this is maybe helping the user get to what they really need/were looking for anyhow. And now that I'm not surfacing tags on the results, they may never notice the bug.

**Comment by liaprins** [comment]
Consider adding # manually in panel for each tag, and then using `ltrim()` on the meta tags for keywords, twitter hashtags, etc.

Then search could be used with original code, I think.

**Comment by liaprins** [comment]
I tried the idea in the last comment, but it would not work. I could not get the tag name to show up (not the article(s)) on the tag page when I clicked a tag that I had made in the panel to be preceded by "#". Only the "#" would show, but not the word itself.

I tried to remove hashes from tag.php template, blogarticle-tag.php snippet, and blog-post.php snippet by deleting them and also with the escape key "\" but it only works in Markdown, so it just make "\" appear before tags. I tried adding the escape key "\" into the tag in the panel itself but it still showed up.

I am fine with the current solution, at least for the time being. This one is not worth the time right now.
